knfc/knfconverter.js

```diff
@@ -0,0 +1,220 @@
+       switch(n.right.s.charAt(0)) {
+          case '!':
+             {
+                ret.add("{" + a + S + b + "}");
+                ret.add("{" + not + a + S + not + b + "}");
+                break;
+          }
+          case '<':
+             {
+                ret.add("{" + not + a + S + not + b + S + c + "}");
+                ret.add("{" + not + a + S + b + S + not + c + "}");
+                ret.add("{" + a + S + b + S + c + "}");
+                ret.add("{" + a + S + not + b + S + not + c + "}");
+                break;
+          }
+          case '-':
+             {
+                ret.add("{" + not + a + S + not + b + S + c + "}");
+                ret.add("{" + a + S + b + "}");
+                ret.add("{" + a + S + not + c + "}");
+                break;
+          }
+          case '%':
+             {
+                ret.add("{" + not + a + S + not + b + S + c + "}");
+                ret.add("{" + not + a + S + not + b + S + not + c + "}");
+                ret.add("{" + a + S + not + b + S + c + "}");
+                ret.add("{" + a + S + b + S + not + c + "}");
+                break;
+          }
+          case '|':
+             {
+                ret.add("{" + not + a + S + b + S + c + "}");
+                ret.add("{" + a + S + not + b + "}");
+                ret.add("{" + a + S + not + c + "}");
+                break;
+          }
+          case '&':
+             {
+                ret.add("{" + a + S + not + b + S + not + c + "}");
+                ret.add("{" + not + a + S + b + "}");
+                ret.add("{" + not + a + S + c + "}");
+                break;
+          }
+       }
+    }
+    return ret;
+ };
+ Abfahrt2_c.generateClauses = function (n)  {
+    var ret = new jv_ArrayList();
+    var help = new Node("<");
+    help.left = new Node("T");
+    help.left.parent = help;
+    help.left.setID(n.id);
+    help.right = new Node(n.s);
+    help.right.parent = help;
+    if (n.left !== null) {
+       help.right.left = new Node(Abfahrt2_c.isClauseChar(n.left.s.charAt(0)) ? "T" : n.left.s);
+       help.right.left.parent = help.right;
+       if (Abfahrt2_c.isClauseChar(n.left.s.charAt(0))) {
+          help.right.left.setID(n.left.id);
+          ret.addAll(Abfahrt2_c.generateClauses(n.left));
+       }
+       ret.add(0, help);
+    }
+    if (n.right !== null) {
+       help.right.right = new Node(Abfahrt2_c.isClauseChar(n.right.s.charAt(0)) ? "T" : n.right.s);
+       help.right.right.parent = help.right;
+       if (Abfahrt2_c.isClauseChar(n.right.s.charAt(0))) {
+          help.right.right.setID(n.right.id);
+          ret.addAll(Abfahrt2_c.generateClauses(n.right));
+       }
+    }
+    return ret;
+ };
+ Abfahrt2_c.isClauseChar = function (c)  {
+    if (arguments.length == 0) return;
+    return c === '!' || c === '<' || c === '-' || c === '|' || c === '&' || c === '%';
+ };
+ Abfahrt2_c.toInternRepresentation = function (s)  {
+    s = s.replaceAll("Â¬", "!");
```

Comment:
schreib hier mal noch überall ein Kommentar dazu, welches Zeichen des tatächlich ist

Reply:
Steht in `toInternRepresentation`